### PR TITLE
Quick read of ipv6 docs and small fixes.

### PR DESCRIPTION
**Pull request by ypid**
* Why on earth does one directly edit the systemd/system/docker.service
  file just to add a start argument?
* Fixed typos.
* I have not fully tested it yet, but I will when I have time.

### CODE REVIEW

File: docs/ipv6.md [hunk 1]
```diff
@@ -37,13 +37,14 @@ If this doesn't work, figure it out.  It may be necessary to add an firewall rul
...
 
-    sed -e 's:^\(ExecStart.*\):\1 --ipv6:' /lib/systemd/system/docker.service | tee /etc/systemd/system/docker.service
+Append the `--ipv6` argument to the `DOCKER_OPTS` variable in:
+
+    /etc/default/docker
 
 Reload the daemon and restart docker so that it takes affect:
 
-    systemctl daemon-reload && systemctl restart docker.service
+    service docker restart
```
Comment (kylemanna) [inline docs/ipv6.md]: If using systemd, people need to use systemctl.  Ubuntu provides `service` binary to manage upstart services and maybe will soon wrap systemd interface.

Probably need service and systemctl commands for upstart and systemd respectively.

**Comment by kylemanna** [comment]
Thanks for reviewing!

I don't understand your comment about modifying the systemd service for two reasons:

1. People shouldn't be modifying the service file directly.  It should be overriden [as described in the init file](https://github.com/kylemanna/docker-openvpn/blob/149cd3a3a34050d761f68e321964fd16c3277bb5/init/docker-openvpn%40.service#L19-L21).  I'm currently running several OpenVPN servers on one of my servers and it works. The goal is to make the systemd service extensible and have sensible defaults.   Open to other suggestions. 

    The IPv6 stuff in the service file is an eye sore, but Docker doesn't handle network configuration very well at this point. 

1. Why are you using the start argument?  Are you storing the state in a single container?  Running a single container makes it very hard to update.  My philosophy is to do a `docker pull` to always pull the latest openvpn image (and fail gracefully if the pull fails) to get the latest security updates and then create a new container with `docker run`.  If you're doing `docker start ...` it suggests you're re-using a container, why?

**Comment by ypid** [comment]
Hi

Not sure if I understand you correctly.

My main confusion was about the `sed -e 's:^\(ExecStart.*\):\1 --ipv6:' /lib/systemd/system/docker.service | tee /etc/systemd/system/docker.service` line which you used to append the `--ipv6` switch. This can be done much simpler and more robust using the `/etc/default/docker` file. (Wrote an Ansible Role for that :smile: [ypid.docker](https://github.com/ypid/ansible-docker))

About the `systemctl restart` vs. `service` command I guess they are interchangeble on most systems running systemd (which is the majority of current distributions). I am just used to the `service` command because it is shorter :smile: (reverted …).

To 2. Not sure what you mean. I published parts of my Docker setup configuration on GitHub. [Here](https://github.com/ypid/docker-makefile/blob/434e00ec2aacaa8c18ff5ce8e2ff9dee386da61b/Makefile#L198-L208) is the relevant part.

BTW. Thanks for your work on v6 integration!

File: docs/ipv6.md [hunk 2]
```diff
@@ -37,13 +37,14 @@ If this doesn't work, figure it out.  It may be necessary to add an firewall rul
...
-Copy the system's existing docker file and append the `--ipv6` argument to the end of the command line:
 
-    sed -e 's:^\(ExecStart.*\):\1 --ipv6:' /lib/systemd/system/docker.service | tee /etc/systemd/system/docker.service
+Append the `--ipv6` argument to the `DOCKER_OPTS` variable in:
+
+    /etc/default/docker
 
 Reload the daemon and restart docker so that it takes affect:
 
-    systemctl daemon-reload && systemctl restart docker.service
```
Comment (kylemanna) [inline docs/ipv6.md]: Still need to tell systemd to examine the updated docker.service file, so the `daemon-reload` is necessary.

**Comment by kylemanna** [comment]
Ahh.  I was thinking of `docker-openvpn@.service` and you're talking about `docker.service`.

The `/etc/default` configuration directory is Debian (and derivative) specific thing, as is `service`.  These will hopefully go away when people finally standardize an init system across distributions.  Systemd defines all these things in a much more extensible way.

In short, we need both until the Debian way dies (as it has already moved ot systemd) and Ubuntu will soon follow.  Fedora and Arch has been on systemd for years.

File: docs/ipv6.md (hunk 2 above)
Comment (ypid) [inline docs/ipv6.md]: v6 is now configured in `/etc/default/docker` which is read on on normal service restart. There should be no need to restart systemd itself for that.

For Step 3 — Setup the systemd Unit File, it is a good idea, but for adding a start augment, no.

**Pull request merged**